test/disassembler.js

```diff
@@ -0,0 +1,101 @@
+'use strict'
+
+var tape = require('tape')
+var disassemble = require('../src/code/disassembler.js').disassemble
+
+tape('Disassembler', function (t) {
+  t.test('empty', function (st) {
+    st.plan(1)
+    st.equal(disassemble(''), '')
+  })
+  t.test('add', function (st) {
+    st.plan(1)
+    st.equal(disassemble('0x01'), 'add')
+  })
+  t.test('push', function (st) {
+    st.plan(1)
+    st.equal(disassemble('0x640203'), '0x0203000000')
+  })
+  t.test('complexcode', function (st) {
+    st.plan(1)
+    var code = '60606040526009600060005055607e8060186000396000f360606040526000357c0100000000000000000000000000000000000000000000000000000000900480630dbe671f146039576035565b6002565b3460025760486004805050604a565b005b6000600090505b600a811015607a5760006000818150548092919060010191905055505b80806001019150506051565b5b5056'
+    var asm = `mstore(0x40, 0x60)
+0x09
+0x00
+pop(0x00)
+sstore
+0x7e
+dup1
+0x18
+0x00
+codecopy
+0x00
+return
+mstore(0x40, 0x60)
+calldataload(0x00)
+0x0100000000000000000000000000000000000000000000000000000000
+swap1
+div
+dup1
+0x0dbe671f
+eq
+0x39
+jumpi
+jump(0x35)
```

Comment:
Yes, but that is the next step. There is no clear way to do it. Perhaps the best way would be to replace all constants that are also jumpdests and put the actual value as a comment.